server/pxf-service/src/main/java/org/greenplum/pxf/service/security/BaseSecurityService.java

```diff
@@ -55,58 +54,52 @@ public BaseSecurityService(SecureLogin secureLogin, UGIProvider ugiProvider) {
+                // we check whether the pxf.service.user.name property was provided
+                // and if provided we use the value as the remote user instead of
+                // the principal defined in pxf.service.kerberos.principal. However,
+                // the principal will need to have proxy privileges on hadoop.
+                String pxfServiceUserName = configuration.get(SecureLogin.CONFIG_KEY_SERVICE_USER_NAME);
+                if (StringUtils.isNotBlank(pxfServiceUserName)) {
+                    serviceUser = pxfServiceUserName;
+                }
             }
-        }
 
-        String remoteUser = (isUserImpersonation ? gpdbUser : serviceUser);
-        String contextId = context.getId();
+            String remoteUser = (isUserImpersonation ? gpdbUser : serviceUser);
 
-        boolean exceptionDetected = false;
-        UserGroupInformation userGroupInformation = null;
-        try {
             // Retrieve proxy user UGI from the UGI of the logged in user
             if (isUserImpersonation) {
-                LOG.debug("{} Creating proxy user = {}", contextId, remoteUser);
+                LOG.debug("Creating proxy user = {}", remoteUser);
                 userGroupInformation = ugiProvider.createProxyUser(remoteUser, loginUser);
             } else {
-                LOG.debug("{} Creating remote user = {}", contextId, remoteUser);
+                LOG.debug("Creating remote user = {}", remoteUser);
                 userGroupInformation = ugiProvider.createRemoteUser(remoteUser, loginUser, isSecurityEnabled);
             }
 
-            LOG.debug("{} Retrieved proxy user {} for server {}", contextId, userGroupInformation, serverName);
-            LOG.debug("Performing request for gpdb_user = {} as [remote_user = {} service_user = {} login_user ={}] with{} impersonation",
+            LOG.debug("Retrieved proxy user {} for server {}", userGroupInformation, serverName);
+            LOG.debug("Performing request for gpdb_user = {} as [remote_user={} service_user={} login_user={}] with{} impersonation",
                     gpdbUser, remoteUser, serviceUser, loginUser.getUserName(), isUserImpersonation ? "" : "out");
             // Execute the servlet chain as that user
             return userGroupInformation.doAs(action);
-        } catch (UndeclaredThrowableException ute) {
-            exceptionDetected = true;
-            // unwrap the real exception thrown by the action
-            throw new IOException(ute.getCause());
-        } catch (InterruptedException ie) {
+        } catch (Exception e) {
             exceptionDetected = true;
-            throw new IOException(ie);
+            throw e;
         } finally {
-            LOG.debug("{} Releasing UGI resources. {}",
-                    contextId, exceptionDetected ? " Exception while processing" : "");
+            LOG.debug("Releasing UGI resources. {}", exceptionDetected ? " Exception while processing" : "");
```

Comment:
adding a period for consistency
```suggestion
            LOG.debug("Releasing UGI resources. {}", exceptionDetected ? " Exception while processing." : "");
```